barman/recovery_executor.py

```diff
@@ -361,17 +364,18 @@ def _setup(self, backup_info, remote_command, dest):
             "recovery_start_time": datetime.datetime.now(dateutil.tz.tzlocal()),
         }
         recovery_info["results"] = results
-
         # Set up a list of configuration files
-        recovery_info["configuration_files"].append("postgresql.conf")
-        if backup_info.version >= 90400:
-            recovery_info["configuration_files"].append("postgresql.auto.conf")
-
-        # Identify the file holding the recovery configuration
-        results["recovery_configuration_file"] = "postgresql.auto.conf"
-        if backup_info.version < 120000:
-            results["recovery_configuration_file"] = "recovery.conf"
-
+        if not recovery_conf_filename:
+            recovery_info["configuration_files"].append("postgresql.conf")
+            if backup_info.version >= 90400:
+                recovery_info["configuration_files"].append("postgresql.auto.conf")
+
+            # Identify the file holding the recovery configuration
+            results["recovery_configuration_file"] = "postgresql.auto.conf"
+            if backup_info.version < 120000:
+                results["recovery_configuration_file"] = "recovery.conf"
+        else:
+            recovery_info["configuration_files"].append(recovery_conf_filename)
```

Comment:
When `recovery_conf_filename` is set it looks like `postgresql.conf` is not going to be added to `recovery_info["configuration_files"]`. This means the config mangling which happens in [_analyse_temporary_config_files](https://github.com/EnterpriseDB/barman/blob/5cdd66eab0aad41d0e1e0b0eac9f4660e6bb85d4/barman/recovery_executor.py#L1196-L1212) will skip the `postgresql.conf` file whenever a custom recovery config file path is used. It feels like we'd be better off always including `postgresql.conf` in `recovery_info["configuration_files"]`, since the recover config file doesn't change the fact that Barman needs to scan `postgresql.conf`.

Also, you should set `results["recovery_configuration_file"]` to `recovery_conf_filename` when it is set - this will make the filename appear in the `barman recover` output. You could also reference it via the `recovery_info["results"]["recovery_configuration_file"]` in the functions which currently have a dedicated `recovery_conf_filename` parameter.

Reply:
I tried to fixed it please check it out

Thanks.